test.py

```diff
@@ -372,6 +374,72 @@ def test_slurm_cfut_dir():
     assert len(os.listdir(cfut_dir)) == 2
 
 
+def test_slurm_max_submit_user():
+    max_submit_jobs = 6
+
+    # MaxSubmitJobs can either be defined at the user or at the qos level
+    for command in ["user root", "qos normal"]:
+        executor = cluster_tools.get_executor("slurm", debug=True)
+        original_max_submit_jobs = executor.get_max_submit_jobs()
+
+        _, _, exit_code = call(
+            f"echo y | sacctmgr modify {command} set MaxSubmitJobs={max_submit_jobs}"
+        )
+        try:
+            assert exit_code == 0
+
+            new_max_submit_jobs = executor.get_max_submit_jobs()
+            assert new_max_submit_jobs == max_submit_jobs
+
+            with executor:
+                futures = executor.map_to_futures(square, range(10))
+
+                result = [fut.result() for fut in futures]
+                assert result == [i ** 2 for i in range(10)]
+
+                job_ids = {fut.cluster_jobid for fut in futures}
+                # The 10 work packages should have been scheduled as 5 separate jobs,
+                # because the cluster_tools schedule at most 1/3 of MaxSubmitJobs at once.
+                assert len(job_ids) == 5
+        finally:
+            _, _, exit_code = call(
+                f"echo y | sacctmgr modify {command} set MaxSubmitJobs=-1"
+            )
+            assert exit_code == 0
+            reset_max_submit_jobs = executor.get_max_submit_jobs()
+            assert reset_max_submit_jobs == original_max_submit_jobs
+
+
+def test_slurm_number_of_submitted_jobs():
+    number_of_jobs = 6
+    executor = cluster_tools.get_executor("slurm", debug=True)
+
+    assert executor.get_number_of_submitted_jobs() == 0
+
+    with executor:
+        futures = executor.map_to_futures(sleep, [1] * number_of_jobs)
+        assert executor.get_number_of_submitted_jobs() == number_of_jobs
+        concurrent.futures.wait(futures)
+
+    assert executor.get_number_of_submitted_jobs() == 0
```

Comment:
I think, the assertion should go into the with-block, as the with-block itself is also waiting for the executor to be finished. To be stricter, doing the assertion after the first waiting would be better imo.